evm/test/Obridge.js

```diff
@@ -0,0 +1,201 @@
+const {
+    time,
+    loadFixture,
+  } = require("@nomicfoundation/hardhat-network-helpers");
+const { anyValue } = require("@nomicfoundation/hardhat-chai-matchers/withArgs");
+const { expect } = require("chai");
+
+describe('OBridge', function () {
+    async function deployOBridge () {
+        const [owner, otherAccount, user, lp] = await ethers.getSigners();
+
+        const OBridge = await hre.ethers.getContractFactory("OBridge");
+        const obridge = await OBridge.deploy()
+        await obridge.deployed();
+
+        return {obridge, owner, otherAccount, user, lp}
+    }
+    async function deployTestERC20Src () {
+        const [owner, otherAccount] = await ethers.getSigners();
+
+        const initialSupply = '10000000000000000000000'
+
+        const TestERC20 = await hre.ethers.getContractFactory("TestERC20Src");
+        const tercSrc = await TestERC20.deploy(initialSupply)
+        await tercSrc.deployed();
+
+        return {tercSrc, owner, otherAccount, initialSupply}
+    }
+    async function deployTestERC20Dst () {
+        const [owner, otherAccount] = await ethers.getSigners();
+
+        const initialSupply = '10000000000000000000000'
+
+        const TestERC20 = await hre.ethers.getContractFactory("TestERC20Dst");
+        const tercDst = await TestERC20.deploy(initialSupply)
+        await tercDst.deployed();
+
+        return {tercDst, owner, otherAccount, initialSupply}
+    }
+
+    let cache = {}
+
+    //TODO Fill use case
```

Comment:
These are commented out. Can we demo a complete exchange using these scripts?

Reply:
In this file are some test cases that are expected to be improved, this part is not related to the main process,
for example, wrong operation data, charging fee, changing owner address, canceling exchange midway, etc.